Remove misleading list of default locations (#10614)
python/mypy: docs/source/config_file.rst

It is important to understand that there is no merging of configuration
files, as it would lead to ambiguity.  The :option:`--config-file <mypy --config-file>` flag
has the highest precedence and must be correct; otherwise mypy will report

> an error and exit.  Without command line option, mypy will look for defaults in the above mentioned order,
> but will use only one of them.

Reviewer: ```suggestion
an error and exit.  Without command line option, mypy will look for configuration files in the above mentioned order.
```
